internal/testing/e2e/service.go

```diff
@@ -23,7 +23,7 @@ import (
 )
 
 const (
-	envoyImage                = "envoyproxy/envoy:v1.21-latest"
+	EnvoyImage                = "envoyproxy/envoy:v1.24-latest"
```

Comment:
This seems fine -- this is pretty much just used in running our dummy services on the mesh, but since they're pulling xDS it's still a necessary change to get this running in test. That said, I'm assuming that changing the actual default value for envoy instances deployed as gateways will happen in a separate PR (that's the one you'll need to regen the golden files for)?

https://github.com/hashicorp/consul-api-gateway/blob/2a3f024662c0130f4cce76806bca69df9ed42efc/internal/k8s/builder/builder.go#L34